docs/cartesian-product.adoc

```diff
@@ -0,0 +1,218 @@
+:page-title: Testing all test parameter combinations (Cartesian product)
+:page-description: Extends JUnit Jupiter with `@CartesianProductTest`, a special `@ParameterizedTest` which tests all combinations of its input
+
+From Wikipedia, the free encyclopedia:
+
+> In mathematics, specifically set theory, the Cartesian product of two sets A and B, denoted A × B, is the set of all ordered pairs (a, b) where a is in A and b is in B.
+> In terms of set-builder notation, that is `A × B = {(a,b) | a ∈ A and b ∈ B}`
+> +[...]+
+> One can similarly define the Cartesian product of n sets, also known as an n-fold Cartesian product, which can be represented by an n-dimensional array, where each element is an n-tuple.
+
+What does all this mean?
+
+The cartesian product of sets is all the possible combinations where you take a single element from each set.
+If you have two sets, `{ 1, 2 }` and `{ 3, 4 }`, their cartesian product is `{ { 1, 3 }, { 1, 4 }, { 2, 3 }, { 2, 4 } }`.
+
+You can use `@CartesianProductTest` and define test parameter sets to run the same test on the cartesian product.
+This means that `@CartesianProductTest` takes the possible test parameter values and runs the test for every possible combination of them.
+
+== Basic Use
+
+`@CartesianProductTest` is used _instead_ of `@Test` or other such annotations (e.g. `@RepeatedTest`).
+The annotation can have a value (see <<Supplying a value to CartesianProductTest>>), the test method can be annotated with `@CartesianValueSource` (see <<Annotating your test method with @CartesianValueSource>>), or the test class can have a static factory method (see <<Writing a static factory method for the parameters>>) providing the arguments.
+
+`@CartesianProductTest` works with sets.
+This means that duplicate elements get removed.
+If your input is `{ 1, 1, 3, 4 }` and `{ 2, 2, 3 }` their cartesian product is `{ { 1, 2 }, { 1, 3 }, { 3, 2 }, { 3, 3 }, { 4, 2 }, { 4, 3 } }`.
+This is because otherwise the test would run with the same parameters multiple times.
+If you need that functionality, you might want to look into https://junit.org/junit5/docs/current/user-guide/#writing-tests-repeated-tests[repeated tests].
+
+== Supplying a value to CartesianProductTest
```

Comment:
I don't like the wording of "a value". I know it's because `value` is the default name of the annotation variable, but "a value" mislead to "a single value", which is false, because you are passing String arrays.

What about something like "Supplying (simple) sets of Strings to the extension"?